core/src/main/java/io/hyperfoil/core/client/netty/Http2Connection.java

```diff
@@ -131,6 +131,19 @@ public void request(HttpRequest request,
 
       ByteBuf buf = bodyGenerator != null ? bodyGenerator.apply(request.session, this) : null;
 
+      if (request.path.contains(" ")) {
+          int question = request.path.indexOf("?");
+          String subFirst = "";
+          String subSecond = "";
+          if (question != -1) {
+              subFirst = request.path.substring(0 , question);
+              subFirst = subFirst.replaceAll(" ", "%20");
```

Comment:
I've suggested replacing that part with `AppendableCharSequence`, was there an issue with that? Using 2x substring, string concatenation and regexp search (!) through `replaceAll` seems rather excessive from memory allocation POV.

Reply:
I will try to use AppendableCharSequence instead today.